src/ext/processObliv/processObliv.ml

```diff
@@ -594,6 +634,12 @@ let rec codegenUncondInstr (instr:instr) : instr = match instr with
         | BNot -> setUnop "__obliv_c__setBitwiseNot" v e loc
         | LNot -> setUnop "__obliv_c__setLogicalNot" v e loc
         end
+    | TFloat(kind, a) when hasOblivAttr a ->
+        begin match op with
+        | Neg -> setUnop "__obliv_c__setNegF" v e loc
+        | BNot -> E.s (E.error "Bitwise operation on float invalid!\n")
```

Comment:
>  GCC does not produce an error; I believe it is just undefined behavior

Are you sure? I just tried it and got `Error: integralPromotion: not expecting float` if I do it inside an obliv-c source file, which may not be the best error message of all time, but it gets the point across. If I do it inside a regular .c file and compile with gcc, I get `error: invalid operands to binary << (have 'float' and 'int')`, which is a pretty good error I think. I'm not sure what the standard says (or fails to say), but it seems the compilers won't allow this behavior regardless.

> we want to discourage the use of bitwise operations on floats in general

While I understand the sentiment here, I personally believe that a highly specific error message has a huge value in the context of debugging, which outweighs any 'general discouragement' a less specific variant may imply. Especially since it seems that both CIL and gcc forbid the shifting of floats anyway (unless I am mistaken).